src/main/java/no/ndla/taxonomy/domain/Metadata.java

```diff
@@ -7,124 +7,90 @@
-    @JoinTable(name = "metadata_grep_code", joinColumns = @JoinColumn(name = "metadata_id"), inverseJoinColumns = @JoinColumn(name = "grep_code_id"))
-    private Set<GrepCode> grepCodes = new HashSet<>();
-
-    // JPA will delete custom field values automatically if they are removed from this set and this
-    // entity persisted.
-    // This is for the @PreRemoval to take effect and automatically remove values when this entity
-    // is deleted - makes sense - right? otherwise service level code would have to always delete values before
-    // deleting this.
-    @OneToMany(cascade = { CascadeType.REMOVE, CascadeType.PERSIST, CascadeType.MERGE,
-            CascadeType.REFRESH }, orphanRemoval = true, mappedBy = "metadata", fetch = FetchType.EAGER)
-    private Set<CustomFieldValue> customFieldValues = new HashSet<>();
-
-    @Column
+    private Set<JsonGrepCode> grepCodes = new HashSet<>();
+
+    private Map<String, String> customFields = new HashMap<>();
+
     private boolean visible = true;
 
+    private EntityWithMetadata parent;
+
     public Metadata() {
-    }
 
-    public Metadata(Metadata metadata) {
-        this.visible = metadata.isVisible();
-        Set<GrepCode> gcSet = new HashSet<>();
-        for (GrepCode code : metadata.getGrepCodes()) {
-            gcSet.add(new GrepCode(code, this));
-        }
-        this.grepCodes = gcSet;
-        Set<CustomFieldValue> cfvSet = new HashSet<>();
-        for (CustomFieldValue customFieldValue : metadata.getCustomFieldValues()) {
-            cfvSet.add(new CustomFieldValue(customFieldValue, this));
-        }
-        this.customFieldValues = cfvSet;
     }
 
-    public void addGrepCode(GrepCode grepCode) {
-        this.grepCodes.add(grepCode);
-
-        if (!grepCode.containsMetadata(this)) {
-            grepCode.addMetadata(this);
-        }
+    public Metadata(EntityWithMetadata parent) {
+        this.parent = parent;
+        this.grepCodes = parent.getGrepCodes();
+        this.visible = parent.getVisible();
+        this.createdAt = parent.getCreatedAt();
+        this.updatedAt = parent.getUpdatedAt();
+        this.customFields = parent.getCustomFields();
     }
 
-    public void removeGrepCode(GrepCode grepCode) {
-        this.grepCodes.remove(grepCode);
+    public Metadata(Metadata metadata) {
+        this.parent = metadata.parent;
+        this.createdAt = metadata.createdAt;
+        this.customFields = metadata.getCustomFields();
+        this.grepCodes = new HashSet<>(metadata.getGrepCodes());
+        this.updatedAt = metadata.updatedAt;
+        this.visible = metadata.isVisible();
+    }
 
-        if (grepCode.containsMetadata(this)) {
-            grepCode.removeMetadata(this);
-        }
+    public Metadata(MetadataDto metadata) {
+        this.customFields = metadata.getCustomFields();
+        this.grepCodes = metadata.grepCodes.stream().map(JsonGrepCode::new).collect(Collectors.toSet());
+        this.visible = metadata.isVisible();
```

Comment:
Når dette settes fra ed så er det ikkje alle felt som sendes inn.
```suggestion
        if (metadata.getCustomFields() != null){
            this.customFields = metadata.getCustomFields();
        }
        if (metadata.getGrepCodes() != null){
            this.grepCodes = metadata.getGrepCodes().stream().map(JsonGrepCode::new).collect(Collectors.toSet());
        }
        if (metadata.isVisible() != null) {
            this.visible = metadata.isVisible();
        }
```
Dette er egentlig ikkje nok, ser det ut som. Ved setting av customField så nullstilles grepCodes.

Reply:
alright, skal titte på det! :+1:

Reply:
Ser at den sletta MetadataService hadde kode for å merge metadata. Kanskje kikke på det.